File: dynamicArray/dynamicarray.py
class DynamicArray:
    def __init__(self, capacity=10):
        self.capacity = capacity
        self.array = [None] * self.capacity
        self.size = 0

    def add(self, element):
        if self.size == self.capacity:
            self.resize_array()
        self.array[self.size] = element
        self.size += 1

    def remove(self, index):
        if index < 0 or index >= self.size:
            print("Invalid index for removal.")
            return

        for i in range(index, self.size - 1):
            self.array[i] = self.array[i + 1]

        self.size -= 1

    def get(self, index):
        if index < 0 or index >= self.size:
            print("Invalid index.")
            return None
        return self.array[index]

    def size(self):
        return self.size

    def resize_array(self):
        self.capacity *= 2
        new_array = [None] * self.capacity
        for i in range(self.size):
            new_array[i] = self.array[i]
        self.array = new_array


if __name__ == "__main__":
    dynamic_array = DynamicArray()

    dynamic_array.add(10)
    dynamic_array.add(20)
    dynamic_array.add(30)

    print("Dynamic Array size:", dynamic_array.size())
    print("Element at index 1:", dynamic_array.get(1))

    dynamic_array.remove(1)

    print("Dynamic Array size after removal:", dynamic_array.size())

File: dynamicArray/DynamicArray.java
public class DynamicArray<T> {
    private Object[] array;
    private int size;
    private int capacity;

    public DynamicArray() {
        capacity = 10;  // Initial capacity
        array = new Object[capacity];
        size = 0;
    }

    public void add(T element) {
        if (size == capacity) {
            resizeArray();
        }
        array[size] = element;
        size++;
    }

    public void remove(int index) {
        if (index < 0 || index >= size) {
            System.out.println("Invalid index for removal.");
            return;
        }
        for (int i = index; i < size - 1; i++) {
            array[i] = array[i + 1];
        }
        size--;
    }

    @SuppressWarnings("unchecked")
    public T get(int index) {
        if (index < 0 || index >= size) {
            System.out.println("Invalid index.");
            return null;
        }
        return (T) array[index];
    }

    public int size() {
        return size;
    }

    private void resizeArray() {
        capacity *= 2;  // Double the capacity
        Object[] newArray = new Object[capacity];
        System.arraycopy(array, 0, newArray, 0, size);
        array = newArray;
    }

    public static void main(String[] args) {
        DynamicArray<Integer> dynamicArray = new DynamicArray<>();

        dynamicArray.add(10);
        dynamicArray.add(20);
        dynamicArray.add(30);

        System.out.println("Dynamic Array size: " + dynamicArray.size());
        System.out.println("Element at index 1: " + dynamicArray.get(1));

        dynamicArray.remove(1);

        System.out.println("Dynamic Array size after removal: " + dynamicArray.size());
    }
}
